[quarkus-next] Aurora CI fails with missing quarkus-bom:999-SNAPSHOT on remote EC2 VM

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

storage

### Describe the bug

Since #47892 (build Quarkus from source), the AuroraDB IT job fails on `quarkus-next`:

```
[ERROR] Non-resolvable import POM: io.quarkus:quarkus-bom:pom:999-SNAPSHOT (absent)
```

My hypothesis (based on the logs and the code) is that because the EC2 Ansible inventory template sets `ansible_become: yes` at the group vars level, it causes `mvn_remote_runner` to execute Maven as root (`HOME=/root`). 
m2 cache (with 999-SNAPSHOT artifacts) is though extracted to `/home/ec2-user/.m2/`.

### Version

nightly

### Regression

- [ ] The issue is a regression

### Expected behavior

Aurora DB job passes on `quarkus-next` branch.

### Actual behavior

https://github.com/keycloak/keycloak/actions/runs/25085594201/job/73500340307

### How to Reproduce?

Run `quarkus-next`.

### Anything else?

_No response_

--- a/.github/scripts/ansible/roles/aws_ec2/templates/inventory.yml.j2
+++ b/.github/scripts/ansible/roles/aws_ec2/templates/inventory.yml.j2
@@ -3,7 +3,6 @@ keycloak:
     {{ cluster_name }}_{{ region  | replace('-','_') }}:
       vars:
         ansible_user: ec2-user
-        ansible_become: yes
         ansible_ssh_private_key_file: {{ cluster_name }}_{{ region }}.pem
       hosts:
 {% for instance in instances.instances %}

--- a/.github/scripts/ansible/roles/keycloak_remote_installer/tasks/install.yml
+++ b/.github/scripts/ansible/roles/keycloak_remote_installer/tasks/install.yml
@@ -41,6 +41,7 @@
   file:
     path: "{{ kc_home }}"
     state: directory
+    owner: "{{ ansible_ssh_user }}"
 
 - name: Install Keycloak src on the remote hosts
   unarchive:

